android-core/src/main/java/com/mparticle/identity/MParticleIdentityClientImpl.java

```diff
@@ -129,6 +165,57 @@ public IdentityHttpResponse modify(IdentityApiRequest request) throws JSONExcept
         return parseIdentityResponse(responseCode, response);
     }
 
+    private void catchRequest(IdentityApiRequest request, IdentityHttpResponse identityHttpResponse, String callType, Long maxAgeTime) throws JSONException {
+        if (mConfigManager.isIdentityCacheFlagEnabled()) {
+            try {
+                if (identityCacheTime <= 0L) {
+                    identityCacheTime = System.currentTimeMillis();
+                    mConfigManager.saveIdentityCacheTime(identityCacheTime);
+                }
+                String key = (request.objectToHash() == null) ? null : request.objectToHash() + callType;
+
+
+                identityCacheArray.put(key, identityHttpResponse);
+                mConfigManager.saveIdentityCache(key, identityHttpResponse);
+                mConfigManager.saveIdentityMaxAge(maxAgeTime);
+            } catch (Exception e) {
+                Logger.error("Exception while processing Identity caching " + e);
+            }
+        }
+    }
+
+    private void clearCatch() {
+        identityCacheArray.clear();
+        mConfigManager.clearIdentityCatch();
+    }
+
+    private IdentityHttpResponse checkIfExists(IdentityApiRequest request, String callType) {
+        if (mConfigManager.isIdentityCacheFlagEnabled()) {
+            try {
+                String key = request.objectToHash() + callType;
+                if (identityCacheTime <= 0L) {
+                    identityCacheTime = mConfigManager.getIdentityCacheTime();
+                }
+                if (maxAgeTimeForIdentityCache <= 0L) {
+                    maxAgeTimeForIdentityCache = mConfigManager.getIdentityMaxAge();
+                }
+                if (identityCacheArray.isEmpty()) {
+                    identityCacheArray = mConfigManager.fetchIdentityCache();
+                }
+                if ((((System.currentTimeMillis() - identityCacheTime) / 1000) <= maxAgeTimeForIdentityCache) && identityCacheArray.containsKey(key)) {
+                    return identityCacheArray.get(key);
+                } else {
+                    return null;
+
+                }
+            } catch (Exception e) {
+                Logger.error("Exception  " + e);
+
+            }
+        }
+        return null;
+    }
```

Comment:
```suggestion
    private IdentityHttpResponse checkIfExists(IdentityApiRequest request, String callType) {
        if (!mConfigManager.isIdentityCacheFlagEnabled()) {
            return null;
        }

        try {
            String key = identityCacheKey(request, callType);
            if (key == null) {
                return null;
            }

            if (identityCacheMap.isEmpty()) {
                identityCacheMap = mConfigManager.fetchIdentityCache();
            }

            IdentityHttpResponse cachedResponse = identityCacheMap.get(key);
            if (cachedResponse != null) {
                if (cachedResponse.getCacheExpirationMillis() > System.currentTimeMillis()) {
                    return cachedResponse;
                } else {
                    // Expired, so remove from cache
                    mConfigManager.removeFromIdentityCache(key);
                }
            }
        } catch (Exception e) {
            Logger.error("Exception  " + e);
        }
        return null;
    }
```

Using the suggestion I made in the `ConfigManager` class, this method would look something like the above.